### ISSUE TITLE
helm delete is deleting all releases

### ISSUE THREAD

**Issue by nirvanagit** [kubernetes/helm#2115]
I have an outage bug here.

While deleting a releases I ran this command:
`helm delete 'app2-test-app-stage-bf1-testeiddccfllhlegfknfreebbikbfhhbkufuejlhbcrrtvn'`

And all my releases were wiped off. The release name I supplied is incorrect, it has some random characters in the end.

On further investigation, I checked the tiller logs and saw:
1. It ran: `helm delete 'app2-test-app-stage-bf1-testeiddccfllhlegfknfreebbikbfhhbkufuejlhbcrrtvn'`
2. Below the above log line it showed:
```2017/03/15 01:20:55 release_server.go:985: uninstall: Deleting app2-test-app-stage-bf1-testeiddccfllhlegfknfreebbikbfhhbkufuejlhbcrrtvn
2017/03/15 01:20:55 release_server.go:913: Executing pre-delete hooks for nginx-test
```
and so on... and it logged that it was deleting all my releases.

3. On further investigation I thought it did helm history of my incorrect release and got all the other releases.
To test it I ran `helm history`
`helm history 'app2-test-app-stage-bf1-testeiddccfllhlegfknfreebbikbfhhbkufuejlhbcrrtvn'`
and it listed all my releases.

Gist: https://gist.github.com/nirvanagit/0099f52fd9bc1843926231d75db1b96d

**Comment by chancez** [comment]
I see similar behavior. If I run the same helm history command _exactly_, I don't get *all* releases, but I get most of them. 

I chose to run the same `helm history` command exactly, but other long names do something similar (details on that below): 
Here's my `helm ls` and `helm history 'app2-test-app-stage-bf1-testeiddccfllhlegfknfreebbikbfhhbkufuejlhbcrrtvn'` output: https://gist.github.com/chancez/423fb37f6da737d850ad461070c85bbd

Interesting things: 
- The revisions are duplicated, which tells me these are from multiple releases, not a single release.
- I only see stuff that's using my `soy-deploy` chart, not `etcd-operator`, or `soy-migrate`. 
- The revisions range from 43 to 146.
  - 43 is the highest revision of the release `commerce-dev-deploy`
  - 146 is the highest revision of the release `tec-production`

I did the fun work, and determined the error occurs when the release name is `64` characters or longer.
